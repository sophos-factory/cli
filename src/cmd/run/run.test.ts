/* eslint-disable @typescript-eslint/no-non-null-assertion */
import { execute } from '../../../tests/helpers/execute';

describe('factoryctl run', () => {
  describe('pipeline', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['run', 'pipeline'], {
          token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });

  describe('job', () => {
    test('throws on missing arguments', async () => {
      await expect(
        execute(['run', 'job'], {
          token: process.env.FACTORY_DYNAMIC_AUTH_TOKEN!
        })
      ).rejects.toMatchSnapshot();
    });
  });
});
