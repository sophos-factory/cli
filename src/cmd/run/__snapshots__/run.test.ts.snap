// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`factoryctl run job throws on missing arguments 1`] = `
"factoryctl run job [job-id]

Executes specified job

Positionals:
  job-id  ID of the job to be run  [string] [default: read from stdin]

Options:
      --version             Show version number  [boolean]
      --address             Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address        Address of the Sophos Factory Auth API server  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token          Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter              Filter output using JSONPath  [string]
      --format              Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose             Print detailed output  [boolean]
      --help                Show help  [boolean]
      --wait                Wait until run execution is finished  [boolean]
      --suppress-events     Suppress run events during pipeline or job run  [boolean]
      --suppress-outputs    Suppress outputs during pipeline or job run  [boolean]
      --suppress-variables  Suppress variables during pipeline or job run  [boolean]
      --var                 Pipeline variable in \`key:value\` format, where value is JSON data or valid JSON object.  [string]
      --project-id          ID of the project containing the job  [string] [required]

Missing required argument: project-id
"
`;

exports[`factoryctl run pipeline throws on missing arguments 1`] = `
"factoryctl run pipeline [pipeline-id]

Executes specified pipeline

Positionals:
  pipeline-id  ID of the pipeline to be executed  [string] [default: read from stdin]

Options:
      --version             Show version number  [boolean]
      --address             Address of the Sophos Factory API server  [string] [default: FACTORY_ADDRESS if set, otherwise https://api.dev.factory.sophos.com/v1]
      --auth-address        Address of the Sophos Factory Auth API server  [string] [default: FACTORY_AUTH_ADDRESS if set, otherwise https://auth.dev.factory.sophos.com/v1]
      --auth-token          Authentication token  [string] [default: FACTORY_AUTH_TOKEN environment variable]
      --filter              Filter output using JSONPath  [string]
      --format              Output format  [choices: "wide", "json", "yaml"] [default: "wide"]
  -v, --verbose             Print detailed output  [boolean]
      --help                Show help  [boolean]
      --wait                Wait until run execution is finished  [boolean]
      --suppress-events     Suppress run events during pipeline or job run  [boolean]
      --suppress-outputs    Suppress outputs during pipeline or job run  [boolean]
      --suppress-variables  Suppress variables during pipeline or job run  [boolean]
      --var                 Pipeline variable in \`key:value\` format, where value is JSON data or valid JSON object.  [string]
      --project-id          ID of the project containing the pipeline  [string] [required]
      --revision-id         ID of the pipeline revision to be executed  [string] [required]

Missing required arguments: project-id, revision-id
"
`;
